### PR TITLE
Populate default TLS config only if files exist

### PR DESCRIPTION
**Pull request by containscafeine**
This commit checks if the certs exist on disk before populating
the default TLS config. This will allow users to set other options
in TLS module without ambassador erroring out because the certs
don't exist.

### CODE REVIEW

**Comment by kflynn** [comment]
I wasn't trying to close this, _and_ I just learned why GitHub didn't want me to delete its original base branch. Sigh. :)

**Comment by kflynn** [comment]
Code looks good -- how can we add an end-to-end test for it?

**Comment by richarddli** [comment]
I tried to follow the epic thread in #469 and added the docs in https://github.com/datawire/ambassador/pull/616.

File: ambassador/ambassador/config.py [hunk 1]
```diff
@@ -1107,12 +1108,16 @@ def generate_intermediate_config(self):
...
+                primary_listener['tls']['ssl_context'] = True
             redirect_cleartext_from = tmod.get('redirect_cleartext_from')
 
         self.envoy_config['listeners'] = [ primary_listener ]
 
         if redirect_cleartext_from:
-            primary_listener['require_tls'] = True
+            # We only want to set `require_tls` on the primary listener when certs are present on the pod
+            if self.tmod_certs_exist(primary_listener['tls']) > 0:
+                primary_listener['require_tls'] = True
```
Comment (containscafeine) [inline ambassador/ambassador/config.py]: @kflynn does this look reasonable?
When we talk about L4 redirection, we want the HTTP requests that arrive at 443 to go right through, for that we don't need `require_tls`.
Does some use-case come to your mind that could break with this change?

File: end-to-end/0-serial/011-websocket/ambassador/service.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+aws:
```
Comment (kflynn) [inline end-to-end/0-serial/011-websocket/ambassador/service.yaml]: Huh. This'll always work in Kubernaut? Hey @plombardi89, is adding an AWS dependency here a good idea right now?

Comment (containscafeine) [reply end-to-end/0-serial/011-websocket/ambassador/service.yaml]: Wait, this file should not even be here. This is auto generated IIRC. Removing.

Comment (containscafeine) [reply end-to-end/0-serial/011-websocket/ambassador/service.yaml]: Removed

File: ambassador/ambassador/config.py (hunk 1 above)
Comment (kflynn) [reply ambassador/ambassador/config.py]: I think this is OK, yeah. Had to think about it for a bit.

**Comment by kflynn** [comment]
My comment for @plombardi89 was addressed by removing the file I was concerned about -- turns out it shouldn't've been included. Taking this one. :smile:

**Pull request merged**